velox/vector/arrow/Bridge.cpp

```diff
@@ -474,15 +476,25 @@ void exportNulls(
     out.null_count = 0;
     return;
   }
-  if (!rows.changed()) {
-    holder.setBuffer(0, vec.nulls());
-    out.null_count = vec.getNullCount().value_or(-1);
-    return;
+
+  auto nulls = vec.nulls();
+
+  // If we're only exporting a subset, create a new validity buffer.
+  if (rows.changed()) {
+    nulls = AlignedBuffer::allocate<bool>(out.length, pool);
```

Comment:
nit: Should the size of this be out.length or vec.size() or rows.end() ? Are we guranteed to have out.length >= vec.size()

Reply:
@kgpai  out.length is set as rows.count() and guaranteed to be smaller than vec.size() (since rows.count() is a subset of it)